### PR TITLE
Support for inheritance chain

### PR DESCRIPTION
**Pull request by Tcharl**
Hi,

Your Issues #13 is nearly resolved :). I'm sorry, but I haven't got much time to contribute to the end.

You need to finish these tasks:
Set the 'dynamic template' in the genmodel to true
Set the template directory to '<project>/templates'

If I can give feedback on the project:

- The company name in the context menu is not a good idea (IHMO), at least you can put it in the 'Help-> about' section
- I didn't resolved the factory generation issue, I hope you will^^.
- If you can convince Ed Merks to generate this way by default (adding some default genmodel properties instead of dialogs), it would be HUGE!

Best regards,

Charlie

### CODE REVIEW

**Comment by opcoach** [comment]
Hi Charlie ! 

That's cool to contribute. I will check it probably next week. 

I saw that you changed the java files generated by the xtend files also.. I will fix it and try it. 

Did you used a specific test case to test it ? I mean a representative ecore model ? 

Thank you !

**Comment by Tcharl** [comment]
Hi Coach!

My reference sample contains a metaclass (sample1), and an other that inherits (sample2).
You can add a methods in the 'sample1' metaclass and see that your override works when calling this method from a 'sample2' instance.

In the genmodel, I just added dynamic template prop to true and referenced <project>/templates in the 'template directory' one.

Then, just have to 'generate dev classe' click, then 'generate model'.

Sorry to not have gitignored xtend-generated java file :S .

**Comment by opcoach** [comment]
Hi !

Oups ! You should not'ignore xtend files :) they are used to generate the java code ... And both xtznd and java files must stand in the projzct (bit xtend files are not then delivered)

I will check it...


Olivier Prouvost
06 28 07 65 64

**Comment by Tcharl** [comment]
I didn't ignored xtend files, but ignored the java files generated by xtend (the xtend-gen folder).

**Comment by opcoach** [comment]
If I compare the new Class.javajet you have provided I can see this diff : 

public<%if (genClass.isAbstract()) {%> abstract<%}%> class 
			<%=genClass.getClassName()%><%=genClass.getTypeParameters().trim()%><% if (!genClass.getClassExtends().contains("MinimalEObjectImpl.Container")){%><%=genClass.getClassExtends().replace("Impl","")%>Impl<%}else{%><%=genClass.getClassExtends()%><%}%><%=genClass.getClassImplements()%>

In other words you compute the class name depending on the default inheritance to MinimalEObjectImpl... but it is not a always the case... right ? 

This implementation is interesting and we should discuss about it. Could you phone me ?

**Comment by Tcharl** [comment]
Hi,

If the class does not inherit from the default Ecore one (MinimalEObjectImpl.Container), i make it inherit from the genModelAddon Overriden one instead of the generated one.

What about an hangout tonight after 20:00 (UTC/GMT +1 Paris)?
